### PR TITLE
DOCS-2313: Make fixups to app client, data client, and motion service package docs

### PR DESCRIPTION
**Pull request by sguequierre**
* Fix code samples, parameter and return descriptions and formatting
* Update descriptions for more info on grabbing organization and dataset ids 
* Add periods and remove "e.g". bc it causes linting error in docs.viam (we aren't supposed to use it)

Note: Motion move_on_globe and move_on_map configuration object formatting: sub-bulleted list -- still looks slightly odd with indenting but couldn't figure out how to completely fix

### CODE REVIEW

File: src/viam/app/data_client.py [hunk 1]
```diff
@@ -773,8 +780,16 @@ async def configure_database_user(self, organization_id: str, password: str) ->
         """Configure a database user for the Viam organization's MongoDB Atlas Data Federation instance. It can also be used to reset the
         password of the existing database user.
 
+        ::
```
Comment (npentrel) [inline src/viam/app/data_client.py:783]: Why is this `::` here? copy paste error?

**Review by npentrel — APPROVED** [review]
LGTM % one small issue

File: src/viam/app/data_client.py (hunk 1 above)
Comment (sguequierre) [reply src/viam/app/data_client.py:783]: @npentrel To start the code sample!

**Review by njooma — APPROVED** [review]
Out of curiosity, why does `e.g.` break docs.viam?

**Comment by sguequierre** [comment]
> Out of curiosity, why does `e.g.` break docs.viam?

It doesn't really break it, it was decided that it wasn't allowed in the docs re. style conventions and the Vale linter raises an error for it that stops merging

**Comment by npentrel** [comment]
@njooma it doesn't break. We've got a linter set up to remind us not to use it, per our style guide. Within the docs community it's a generally accepted convention to move away from latin abbreviations for accessibility reasons, so it's a good change to make across the board

**Review by stuqdog — APPROVED** [review]

**Pull request merged**